.github/workflows/update-helm-charts-index.yml

```diff
@@ -0,0 +1,67 @@
+name: hashicorp/waypoint-helm/update-helm-charts-index
+on:
+  push:
+    tags:
+      - 'v[0-9]+.[0-9]+.[0-9]+*'
+jobs:
+  update-helm-charts-index:
+    outputs:
+      go-version: ${{ steps.get-go-version.outputs.version }}
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@ac593985615ec2ede58e132d2e21d2b1cbd6127c # v3.3.0
+      - name: Determine Go version
+        id: get-go-version
+        run: |-
+          echo "Building with Go $(cat ".go-version")"
+          echo "go-version=$(cat ".go-version")" >> "$GITHUB_OUTPUT"
+      - name: Setup Go
+        uses: actions/setup-go@6edd4406fa81c3da01a34fa6f6343087c207a568 # v3.5.0
+        with:
+          go-version: ${{ steps.get-go-version.outputs.go-version }}
+      - name: verify Chart version matches tag version
+        run: |-
+          GO111MODULE=on go install github.com/mikefarah/yq/v2@latest
+          #git_tag=$(echo "${{ github.ref_name }}" | sed 's/v//g')
+          git_tag=0.1.18
+          chart_tag=$(yq r Chart.yaml version)
+          if [ "${git_tag}" != "${chart_tag}" ]; then
+            echo "chart version (${chart_tag}) did not match git version (${git_tag})"
+            exit 1
+          fi
+      - name: update helm-charts index
+        id: update
+        env:
+          # FIXME: waypoint team to input token
+          GH_TOKEN: ${{ secrets.HELM_CHARTS_GITHUB_TOKEN }}
+        run: |-
+          gh workflow run publish-charts.yml \
+            --repo hashicorp/helm-charts \
+            --ref main \
+            -f SOURCE_TAG="${{ github.ref_name }}" \
+            -f SOURCE_REPO="${{ github.repository }}"
+      - name: Prepare Slack Notification
+        if: always()
+        run: |
+          case "${{ steps.update.conclusion }}" in
+            failure)
+              echo "SLACK_MESSAGE=:red_circle: The vault-helm charts index update failed." >> "$GITHUB_ENV"
+              echo "SLACK_BLOCK_COLOR=ed5c5c" >> "$GITHUB_ENV"
+              ;;
+            success)
+              echo "SLACK_MESSAGE=:tada: The vault-helm charts index update succeeded." >> "$GITHUB_ENV"
+              echo "SLACK_BLOCK_COLOR=1cbf43" >> "$GITHUB_ENV"
+              ;;
+          esac
+      - name: Send Slack Notification
+        # note: this will skip Slack notification if the job is skipped or cancelled.
+        if: ${{ (success() || failure()) && env.SLACK_MESSAGE != '' }}
+        uses: ./.github/actions/slack-status
+        with:
+          block-color: ${{ env.SLACK_BLOCK_COLOR }}
+          message: ${{ env.SLACK_MESSAGE }}
+          run-url: ${{ env.RUN_URL }}
+          # FIXME: waypoint team to input token
+          slack-webhook-url: ${{ secrets.SLACK_WEBHOOK_URL }}
```

Comment:
We need the `Waypoint` team to update this secret in this repository using these guidelines: https://github.com/slackapi/slack-github-action#setup

Reply:
Done!